packages/core/src/agent/didcomm/versions/v2/sicpa-didcomm/DIDCommV2EnvelopeService.ts

```diff
@@ -0,0 +1,95 @@
+import type { AgentContext } from '../../../../context'
+import type { DecryptedMessageContext, EncryptedMessage, SignedMessage } from '../../../types'
+import type { DIDCommV2Message } from '../DIDCommV2Message'
+import type { DIDCommV2EnvelopeService, PackMessageParams } from '../index'
+import type { default as didcomm, IMessage } from 'didcomm'
+
+import { Key } from '../../../../../crypto'
+import { AriesFrameworkError } from '../../../../../error/AriesFrameworkError'
+import { injectable } from '../../../../../plugins'
+import { JsonEncoder } from '../../../../../utils'
+import { AgentConfig } from '../../../../AgentConfig'
+import { DIDCommMessageVersion, MessageType } from '../../../types'
+
+import { DIDResolverService } from './DIDResolverService'
+import { SecretResolverService } from './SecretResolverService'
+
+@injectable()
+export class DIDCommV2SicpaEnvelopeService implements DIDCommV2EnvelopeService {
```

Comment:
Can we move everything related to the sicpa didcomm library into a separate package (can be in this repo) and only keep interfaces in  AFJ that are generic (and e.g. could be used to implement DIDComm v2 with Askar).

Ideally we have a fully generic DidCommV2Enveloppe implementation (just like we have with DIDComm V1) and only the packing is handled by the wallet. Not sure if this is possible with the sicpa didcomm library though